wrapanapi/systems/rhevm.py

```diff
@@ -0,0 +1,945 @@
+            size: integer size of disk in bytes, ex 8GB: 8*1024*1024
+            interface: string disk interface type
+            format: string disk format type
+            active: boolean whether the disk is active
+        Returns: None
+        Notes:
+            Disk format and interface type definitions, and their valid values,
+            can be found in ovirtsdk documentation:
+            http://ovirt.github.io/ovirt-engine-sdk/4.1/types.m.html#ovirtsdk4.types.DiskInterface
+            http://ovirt.github.io/ovirt-engine-sdk/4.1/types.m.html#ovirtsdk4.types.DiskFormat
+        """
+        disk_attachments_service = self.api.disk_attachments_service()
+        disk_attachment = disk_attachments_service.add(
+            types.DiskAttachment(
+                disk=types.Disk(
+                    format=types.DiskFormat(format),
+                    provisioned_size=size,
+                    storage_domains=[
+                        types.StorageDomain(
+                            name=storage_domain,
+                        )
+                    ]
+                ),
+                interface=types.DiskInterface(interface),
+                active=active
+            )
+        )
+        wait_for(self._is_disk_ok, func_args=[disk_attachment.disk.id], delay=5, num_sec=900,
+                 message="check if disk is attached")
+
+    def connect_direct_lun(self, lun_name, lun_ip_addr, lun_port, lun_iscsi_target):
+        """
+        Connects a direct lun disk to the VM.
+
+        Args:
+            lun_name: name of LUN
+            lun_ip_addr: LUN ip address
+            lun_port: LUN port
+            lun_iscsi_target: iscsi target
+        """
+        disk_attachments_service = self.api.disk_attachments_service()
+        if not self.system.does_disk_exist(lun_name):
+            disk_attachment = types.DiskAttachment(
+                disk=types.Disk(
+                    name=lun_name,
+                    shareable=True,
+                    format='raw',
+                    lun_storage=types.HostStorage(
+                        type=types.StorageType.ISCSI,
+                        logical_units=[
+                            types.LogicalUnit(
+                                address=lun_ip_addr,
+                                port=lun_port,
+                                target=lun_iscsi_target,
+                            )
+                        ]
+                    )
+                ),
+                interface=types.DiskInterface.VIRTIO,
+                active=True
+            )
+        else:
+            disk_attachment = self._get_disk_attachment_service(lun_name).get()
+        disk_attachments_service.add(disk_attachment)
+        wait_for(
+            self._is_disk_ok, func_args=[disk_attachment.disk.id], delay=5, num_sec=900,
+            message="check if disk is attached"
+        )
+        return True
+
+    def disconnect_disk(self, disk_name):
```

Comment:
I didn't find any usage of it and it always returns True.

Reply:
Adding wait_for so it'll wait for the disk to be detached from the VM before returning True